rfcs/0002-mvp.md

```diff
@@ -0,0 +1,159 @@
+
+
+### Migration questions
+[migration-questions]: #migration-questions
+
+The `supa migrate` task is the trickiest part. How do we determine what migrations should run on our database, since the database is created declaratively? We would need to compare the databases somehow, and figure out the `up/down` migrations. Alternatively we could do "stateless" migrations, where we just run `restore` on a brand new database, then re-route traffic, but then how do we get the user's prod data into the new database. Bear in mind that enterprise users might have terrabytes of data, and so copying it across will be slow. 
+
+
+### Other questions
+- what is the prefix? should it be `supabase` (clearer) or `supa` (might have a few people calling it "supper/suppa")
+- Which CLI tool/framework/libraries should we use?
+- `supa start` requires docker-compose. Is it reasonable to expect that everyone will have docker set up on their computer? Can we make this easier? 
+- How much functionality should we push to postgres-meta? 
+
+
+
+# Future work
+[future]: #future-work
+
+### Supabase platform 
+
+```sh
+supa deploy   # Deploy a local app to prod. On the first time this is run, it should generate a `./supabase.js` with relevant Supabase config.
+supa orgs     # Manage orgs
+supa project  # Manage projects
+supa logs     # Fetch logs
+supa scale    # Scale the current database
+...etc
+```
+
+
+### Generators (bonus)
+
+Since I have been using typescript in one of our repos, I have discovered a lot of limitations around our [Typescript Generators](https://supabase.io/docs/client/generating-types). This is because Open API spec doesn't have all the tables (if they aren't exposed to `postgres` role), and sometimes it expects some inputs where the database already has defaults (for example our OpenAPI for `created_at` expects a string, when it is actually auto-generated).
+
+I propose here a few ideas for generators. While they are exposed through the CLI, I think we should actually build these into [pg-api](https://github.com/supabase/pg-api), so that anyone can use them without the supabase CLI. (I also think this should be the case with the "dump" functionality).
+
+```sh
+supa gen json api # Dumps the JSON Schema for the API
+supa gen json db # generate AJV schemas for input validation
+
+supa gen typescript api  # generate typescript interfaces for api
+supa gen typescript db # generate typescript interfaces for database. Note that these types should have an interface for each table with "insert types" and "select types". These are actually different because the database can generate default values, so some fields aren't "required" 
+
+
+# Models: 
+# This is my suggestion for a "local first" database which has all of the rules/relationships we need to build a a realtime user store with automatic synchronization via our Realtime engine.
+# A bit like https://github.com/mobxjs/mst-gql (but hopefully much simpler)
+supa gen typescript store 
```

Comment:
@joshnuss 's ActiveRecord library could be a good starting place for JS: 
https://github.com/joshnuss/supabase-active-record